testing_and_setup/atmosphere/setup_run_dir.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+
+BLUE = '\033[34m'
+GREEN = '\033[32m'
+RED = '\033[31m'
+
+def color_text(color, t):
+    import os
+
+    if os.isatty(1):
+        return color + t + '\033[0m'
+    else:
+        return t
+
+
+def print_color(color, t):
+    print(color_text(color, t))
+
+
+def are_same_files(a, b):
+    import filecmp
+
+    return filecmp.cmp(a, b, shallow=True)
+
+
+def copy_file(src, dst, overwrite):
+    import shutil
+    import os
+
+    if os.path.exists(dst):
+        if are_same_files(src, dst):
+            print_color(BLUE, os.path.basename(dst) + ' already exists. No need to copy it.')
+            return
+        elif not overwrite:
+            print_color(RED, dst + ' already exists, but it differs')
+            print_color(RED, '  from ' + src + ' .')
+            print_color(RED, '  ' + os.path.basename(dst) + ' will not be copied.')
+            print_color(RED,'  Use the -o/--overwrite option to copy this file anyway.')
+            return
+
+    try:
+        shutil.copyfile(src, dst)
+        print_color(GREEN, 'Copying ' + os.path.basename(dst))
+    except:
+        print_color(RED,'Error: Could not copy ' + src + ' to ' + dst)
+
+
+def link_file(src, dst, overwrite):
+    import os
+
+    if os.path.exists(dst):
+        if are_same_files(src, dst):
+            print_color(BLUE, os.path.basename(dst) + ' already exists. No need to link it.')
+            return
+        elif not overwrite:
+            print_color(RED, dst + ' already exists, but it differs')
+            print_color(RED, '  from ' + src + ' .')
+            print_color(RED, '  ' + os.path.basename(dst) + ' will not be linked.')
+            print_color(RED, '  Use the -o/--overwrite option to link this file anyway.')
+            return
+        else:
+            os.remove(dst)
+
+    try:
+        os.symlink(src, dst)
+        print_color(GREEN, 'Linking ' + os.path.basename(dst))
+    except:
+        print_color(RED, 'Error: Could not link ' + src +' to '+ dst)
+
+
+def missing_files(mpas_root, required_files):
+    import os
+
+    file_list = []
+    for f in required_files:
+        if not os.path.isfile(mpas_root + '/' + f):
+            file_list.append(f)
+
+    if len(file_list) == 0:
+        return None
+    else:
+        return file_list
+
+
+if __name__ == '__main__':
+    import os
+    import sys
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('run_dir', help='the run directory to set up')
+    parser.add_argument('-o', '--overwrite', action='store_true', help='overwrite any existing files and symbolic links')
```

Comment:
PEP8 also recommends a max line length of 79 chars. Suggested, but fine to leave it as is. https://peps.python.org/pep-0008/#maximum-line-length

Reply:
I've wrapped some lines, though I'm inclined to leave others (mostly lines that involve long string literals) as is.